### PR TITLE
fix: ensure release is called

### PR DESCRIPTION
**Pull request by robertsLando**
Fixes: https://github.com/moscajs/aedes/issues/666

The `_release` should be called after setting the id to 0 otherwise this could cause stucks like the one in the linked PR.

When this happens the strict check `++i === this._count` fails as i isn't resetted yet so the release is never called and this stucks everything...

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
```
Comment (mcollina) [inline .github/workflows/ci.yml:11]: I think this is missing quite a few Node.js versions.

Comment (mcollina) [reply .github/workflows/ci.yml:11]: Yes please, thx

**Comment by robertsLando** [comment]
@JaosnHsieh The problem is that `reusify` doesn't create a new instance of `NoResultsHolder` it instead re-use the first one created, for this reason not only instance vars (`this.`) must be cleared but also internal vars (like `var i`), in the case of `NoResultsHolder` the `_release` was called before resetting `i = 0` and in cases where the `release` function specified in parallel constructor re-calls a new parallel (like in mqemitter, where release calls `_do` that makes the parallel call) this could create a race condition and sometimes the index isn't resetted making the check `++i === count` to fail and causes the stuck

In my test case I wasn't able to reproduce this exact issue but if you remove the `i = 0` it fails.

**Pull request merged**